### PR TITLE
Merge TopLevel and MethodBody speculative models

### PR DESCRIPTION
**Pull request by drognanar**
Combines top level and method body semantic model classes.
In effect the speculative model for top level code is not null.
This enables top level analysis for top level scope such as type simplifications.

### CODE REVIEW

**Comment by gafter** [comment]
@dotnet/roslyn-compiler @AlekseyTs Can you please review this proposed code simplification?

**Comment by drognanar** [comment]
FYI @dotnet/roslyn-interactive

**Comment by amcasey** [comment]
FYI @mavasani for the speculative semantic model

**Comment by amcasey** [comment]
What does `TopLevelCodeBinder` do differently from `MethodBodyBinder`?  It looks redundant.

**Comment by amcasey** [comment]
This seems like a reasonable way to unblock further investigations, but it feels like more investigation is required to determine how and why the VB implementation differs from the C# implementation.  In particular, it appears that C# binding is based on the containing type whereas VB binding is based on the containing method (the script constructor?).  If we're going to merge this PR (probably after eliminating `TopLevelCodeBinder`), I think it would make sense to create a work item to review the handling of top-level statements in the binder and semantic model.

**Comment by mavasani** [comment]
@amcasey @drognanar The change by itself should work fine as top level code is essentially executable code within implicitly declared symbol. However, you probably need lot more testing to verify that the current [implementation](http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Binding/MethodBodySemanticModel.vb,35)which computes the speculative model, and subsequent calls into the returned speculative model will work for all top level cases.

File: src/Compilers/VisualBasic/Portable/Binding/MethodBodySemanticModel.vb [hunk 1]
```diff
@@ -16,7 +16,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' <summary>
         ''' Creates an MethodBodySemanticModel that allows asking semantic questions about an attribute node.
         ''' </summary>
-        Friend Shared Function Create(binder As MethodBodyBinder, Optional ignoreAccessibility As Boolean = False) As MethodBodySemanticModel
+        Friend Shared Function Create(binder As SubOrFunctionBodyBinder, Optional ignoreAccessibility As Boolean = False) As MethodBodySemanticModel
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Binding/MethodBodySemanticModel.vb]: What is the motivation behind this change?

Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Binding/MethodBodySemanticModel.vb]: Simplifying the compiler by eliminating an unnecessary type. This changed line reflects the use of the base type instead of the type that has been eliminated.

**Comment by gafter** [comment]
LGTM.

@dotnet/roslyn-compiler Can I please have a second sign-off?

**Comment by AlekseyTs** [comment]
LGTM, assuming an appropriate testing was performed.
